src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp

```diff
@@ -0,0 +1,35 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <type_traits>
+
+/*!
+ * \ingroup TypeTraitsGroup
+ * \brief Generate a metafunction that will retrieve the specified
+ * `static constexpr` member variable, or, if not present, a default.
+ *
+ * \note This macro requires the type traits created by
+ * `CREATE_HAS_STATIC_MEMBER_VARIABLE` and
+ * `CREATE_HAS_STATIC_MEMBER_VARIABLE_V` but does not create them --
+ * both of those macros should be called before this macro to generate
+ * the needed type traits.
+ */
+#define CREATE_GET_STATIC_MEMBER_VARIABLE_OR_DEFAULT(ALIAS_NAME)          \
+  template <typename CheckingType, auto Default,                          \
+            bool present = has_##ALIAS_NAME##_v<CheckingType>>            \
```

Comment:
Do you actually need `has_ALIAS_NAME_v`? I think you can replace this line with `typename = std::void_t<>` and in the specialization below replace the bool with `std::void_t<CheckingType::ALIAS_NAME>`, or am I missing something?

Reply:
If I were designing the interface, I would have done it that way, but I'm not going to redo existing interfaces in this PR.

(You can use decltype to turn the value into a type to SFINAE with.)